src/net.cpp

```diff
@@ -348,6 +348,11 @@ CNode* CConnman::FindNode(const CService& addr)
     return nullptr;
 }
 
+bool CConnman::AlreadyConnectedToPeer(const CAddress& addr)
+{
+    return FindNode(static_cast<CNetAddr>(addr)) || FindNode(addr.ToStringIPPort());
```

Comment:
Note that this is inefficient (lock cs_vNodes, iterate through nodes, unlock, lock, iterate, lock), but that was a problem with the existing code. It could be made more efficient by converting this to a `ForEachNode()` call with a lambda returning if the address matches the `CNetAddr` or string.

Reply:
Agreed, but I didn't want to rewrite this.

Reply:
That's reasonable. It's not a change in behaviour. Marking resolved.